src/fsw/FCCode/OrbitController.cpp

```diff
@@ -120,32 +139,57 @@ void OrbitController::execute() {
 }
 
 double OrbitController::time_till_node(double theta, const lin::Vector3d &pos, const lin::Vector3d &vel) {
+
+
+    // Collect Relative Orbit Estimator State
+    unsigned char rel_orbit_state=rel_orbit_valid_fp->get();
+
     // Calculate angular velocity (w = v/r)
     double ang_vel = lin::norm(vel)/lin::norm(pos);
 
     // Calculate the times until each node (theta_node = theta_now + w*t)
     double min_time = std::numeric_limits<double>::max();
-    for (size_t i=0; i < 3; i++) {
-        double time_til_node = (firing_nodes[i] - theta) / ang_vel;
-        if (time_til_node > 0 && time_til_node < min_time) {
-            min_time = time_til_node;
+
+    auto next_node = [&](auto const &arr) -> double {
+        for (auto i = 0; i < arr.size(); i++) {
+            double time_til_node = (arr[i] - theta) / ang_vel;
+            if (time_til_node > 0 && time_til_node < min_time) {
+                min_time = time_til_node;
+            }
```

Comment:
Keeping `min_time` outside of the lambda but then still returning it is, at the very least, weird and we should move it into the local scope something like this:
```suggestion
    auto next_node = [&](auto const &arr) -> double {
        auto min_time = std::numeric_limits<double>::max();
        for (auto const node : arr) {
            auto const time_til_node = (node - theta) / ang_vel;
            if (time_til_node > 0 && time_til_node < min_time) {
                min_time = time_til_node;
            }
```
Also, we can use a range based for loop if we want! Just don't forget to remove line 127/151 if you go with this change.